app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,16 @@
+class LikesController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+
+    @like = current_user.likes.new(
+      author_id: current_user.id,
+      post_id: @post.id
+    )
+
+    if @like.save
+      redirect_to user_post_path(current_user, @post)
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+end
```

Comment:
- Please there should be a route in the `config/routes.rb` file for creating like

**NB: This comment was supposed to be lined with `Route` file, unfortunately, you did not make the file visible**